shard-operator-parent/shard-operator-v2/src/main/java/com/redhat/service/smartevents/shard/operator/v2/controllers/ManagedBridgeController.java

```diff
@@ -72,100 +65,72 @@ public class ManagedBridgeController implements Reconciler<ManagedBridge>,
     @Inject
     OperatorMetricsService metricsService;
 
+    @Inject
+    ManagerClient managerClient;
+
     @Override
     public UpdateControl<ManagedBridge> reconcile(ManagedBridge managedBridge, Context context) {
 
         LOGGER.info("Reconciling ManagedBridge: '{}' in namespace '{}'",
                 managedBridge.getMetadata().getName(),
                 managedBridge.getMetadata().getNamespace());
 
-        ManagedBridgeStatus status = managedBridge.getStatus();
-
-        if (!status.isReady() && isTimedOut(status)) {
-            //TODO - Callback to Manager will be added in https://issues.redhat.com/browse/MGDOBR-1267
-            //            notifyManagerOfFailure(bridgeIngress,
-            //                                   new ProvisioningTimeOutException(String.format(ProvisioningTimeOutException.TIMEOUT_FAILURE_MESSAGE,
-            //                                                                                  bridgeIngress.getClass().getSimpleName(),
-            //                                                                                  bridgeIngress.getSpec().getId())));
-            status.markConditionFalse(ConditionTypeConstants.READY);
-            return UpdateControl.updateStatus(managedBridge);
-        }
-
-        Secret secret = managedBridgeService.fetchBridgeSecret(managedBridge);
+        try {
+            ManagedBridgeStatus status = managedBridge.getStatus();
 
-        if (secret == null) {
-            LOGGER.info("The Secret for the ManagedBridge with id '{}' has been not created yet.",
-                    managedBridge.getSpec().getId());
-            if (!status.isConditionTypeFalse(ConditionTypeConstants.READY)) {
-                status.markConditionFalse(ConditionTypeConstants.READY);
-            }
-            if (!status.isConditionTypeFalse(ManagedBridgeStatus.SECRET_AVAILABLE)) {
+            Secret secret = managedBridgeService.fetchBridgeSecret(managedBridge);
+            if (secret == null) {
                 status.markConditionFalse(ManagedBridgeStatus.SECRET_AVAILABLE);
+                throw new ReconciliationException();
```

Comment:
Not sure what you want to do with this `throw new ReconciliationException()`. I have few comments: 
1) The fact that the a Deployment or a Broker is not ready is not an exception: it is actually expected that these resources are not ready immediately like a ConfigMap. As this is not an Exception, our code should handle it properly with no Exceptions - unless you want to use Exception to control the flow, which I would discourage as well. 
2) more importantly, you are updating the conditions and then you `throw` the exception: the CRD on k8s is not going to be updated as you would expect. 
3) The operator is going to raise tons of `Error during event processing ExecutionScope...` exceptions
4) After a given number of retries (configurable in the operator sdk), the operator sdk will raise a `Exhausted retries for ExecutionScope` exception and will stop retrying.

Reply:
As far as I see the design choice of that exception is like a "goto" to stop execution and reschedule the loop in a single `catch`. The exception is [thrown inside a `try/catch`](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/62a73d022687997341dc3a0b9d990b3fe2611757/shard-operator-parent/shard-operator-v2/src/main/java/com/redhat/service/smartevents/shard/operator/v2/controllers/ManagedBridgeController.java#L124) and not "leaked" outside that method

Reply:
Thanks @danielezonca , I missed that. I'm personally very against using Exceptions for flow control: they are expensive and the GOTO-like effect is something that should be avoided in general.

Reply:
With exception I could write handling of failure in single place inside catch block.

Reply:
If we do perform the refactor [mentioned here](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/1438#discussion_r1050500887) to avoid manager notify directly in the loop we don't need an exception just to centralise this logic, we can still keep logic in a single method like
```java
private UpdateControl<ManagedBridge> handleFailure(ManagedBridge managedBridge) {
    return UpdateControl.updateStatus(managedBridge).rescheduleAfter(reconcileIntervalMillis);
}
```
and replace all `throw new ReconciliationException()` with
```java
if (path == null) {
    LOGGER.info("The Knative Broker Resource for ManagedBridge '{}' in namespace '{}' is not ready",
            managedBridge.getMetadata().getName(),
            managedBridge.getMetadata().getNamespace());
    status.markConditionFalse(ManagedBridgeStatus.KNATIVE_BROKER_AVAILABLE);
    return handleFailure(managedBridge);
}
```

It is the same and we avoid to throw an exception. This makes the solution also a bit less error prone because `ReconciliationException` must be catch while now we can add new failure/stop condition without the need to enforce the try/catch

@vaibhavjainwiz Wdyt? I agree that we should not use exceptions as "goto"

Reply:
+1 to @danielezonca comment.

Reply:
done.